x-pack/plugins/security_solution/server/usage/endpoints/fleet_saved_objects.ts

```diff
@@ -5,38 +5,40 @@
  * 2.0.
  */
 
-import { ISavedObjectsRepository } from 'src/core/server';
+import { ElasticsearchClient, SavedObjectsClientContract } from 'src/core/server';
+import { AgentService } from '../../../../fleet/server';
 import { AgentEventSOAttributes } from './../../../../fleet/common/types/models/agent';
 import {
   AGENT_SAVED_OBJECT_TYPE,
   AGENT_EVENT_SAVED_OBJECT_TYPE,
 } from './../../../../fleet/common/constants/agent';
-import { Agent, defaultPackages as FleetDefaultPackages } from '../../../../fleet/common';
+import { defaultPackages as FleetDefaultPackages } from '../../../../fleet/common';
 
 export const FLEET_ENDPOINT_PACKAGE_CONSTANT = FleetDefaultPackages.Endpoint;
 
-export const getFleetSavedObjectsMetadata = async (savedObjectsClient: ISavedObjectsRepository) =>
-  savedObjectsClient.find<Agent>({
-    // Get up to 10000 agents with endpoint installed
-    type: AGENT_SAVED_OBJECT_TYPE,
-    fields: [
-      'packages',
-      'last_checkin',
-      'local_metadata.agent.id',
-      'local_metadata.host.id',
-      'local_metadata.host.name',
-      'local_metadata.host.hostname',
-      'local_metadata.elastic.agent.id',
-      'local_metadata.os',
-    ],
-    filter: `${AGENT_SAVED_OBJECT_TYPE}.attributes.packages: ${FLEET_ENDPOINT_PACKAGE_CONSTANT}`,
+export const getFleetSavedObjectsMetadata = async (
+  savedObjectsClient: SavedObjectsClientContract,
+  agentService: AgentService | undefined,
+  esClient: ElasticsearchClient
+) => {
+  const agentData = await agentService?.listAgents(savedObjectsClient, esClient, {
+    showInactive: true,
     perPage: 10000,
     sortField: 'enrolled_at',
     sortOrder: 'desc',
+    kuery: `${AGENT_SAVED_OBJECT_TYPE}.attributes.packages: ${FLEET_ENDPOINT_PACKAGE_CONSTANT}`,
   });
+  return agentData;
+};
+
+/*
+  TODO: AS OF 7.13, this access will no longer work due to the enabling of fleet server. An alternative route will have
```

Comment:
This will break in 7.13

Reply:
Should we also put this information in an issue?

When it breaks in 7.13 will it have an adverse affect on the plugin? Or will it just log and fail silently?

Reply:
We currently don't log to prevent accidental leaking of PII, but yea if it stays as is for 7.13, we'll only get the `total_installed` and `os` details. Everything else will fail silently. Yea, I'm gonna add it into the issue that @nchaulet created that's referenced in the description here.

Reply:
@jonathan-buttner actually went with a comment [here](https://github.com/elastic/endpoint-dev/issues/8382#issuecomment-794445517) to avoid splitting the discussion